sdk/eventhub/event-hubs/src/linkEntity.ts

```diff
@@ -162,9 +178,21 @@ export class LinkEntity {
       this.name,
       this.address
     );
-    await defaultLock.acquire(this._context.negotiateClaimLock, () => {
-      return this._context.cbsSession.negotiateClaim(this.audience, tokenObject.token, tokenType);
-    });
+    await defaultCancellableLock.acquire(
+      this._context.negotiateClaimLock,
+      () => {
+        return this._context.cbsSession.negotiateClaim(
+          this.audience,
+          tokenObject.token,
+          tokenType,
+          { abortSignal }
+        );
+      },
+      {
+        abortSignal,
+        timeoutInMs: getRetryAttemptTimeoutInMs(undefined)
```

Comment:
Shouldnt this be `timeoutInMs - timeTakenByInit` ?

Reply:
Updated.